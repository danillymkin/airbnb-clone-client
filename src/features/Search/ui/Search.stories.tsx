import { Meta, StoryObj } from '@storybook/react';
import { Search } from './Search';

const meta: Meta = {
  title: 'Features/Search',
  component: Search,
};

export default meta;

type Story = StoryObj<typeof Search>;

export const Default: Story = {
  render: () => <Search />,
};
