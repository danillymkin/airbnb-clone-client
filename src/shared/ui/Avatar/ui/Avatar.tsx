import React, { FunctionComponent } from 'react';
import Image from 'next/image';

export const Avatar: FunctionComponent = (): JSX.Element => {
  return (
    <Image
      className={'rounded-full'}
      src={'/images/avatar-fallback.jpg'}
      width={30}
      height={30}
      alt={'Avatar'}
    />
  );
};
