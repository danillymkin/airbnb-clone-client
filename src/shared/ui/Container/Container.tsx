import React, { FunctionComponent } from 'react';

interface ContainerProps {
  children: React.ReactNode;
}

type Props = ContainerProps;

export const Container: FunctionComponent<Props> = ({
  children,
}): JSX.Element => {
  return (
    <div
      className={'lg: mx-auto max-w-[2520px] px-4 sm:px-2 md:px-10 xl:px-20'}
    >
      {children}
    </div>
  );
};
