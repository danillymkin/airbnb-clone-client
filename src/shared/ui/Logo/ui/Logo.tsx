'use client';

import React, { FunctionComponent } from 'react';
import Image from 'next/image';

export const Logo: FunctionComponent = (): JSX.Element => {
  return (
    <Image
      alt={'Logo'}
      className={'block cursor-pointer'}
      height={31}
      width={100}
      src={'/images/logo.png'}
      priority
    />
  );
};
