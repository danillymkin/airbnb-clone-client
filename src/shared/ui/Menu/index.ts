export { Menu } from './ui/Menu/Menu';
export { MenuItem } from './ui/MenuItem/MenuItem';
