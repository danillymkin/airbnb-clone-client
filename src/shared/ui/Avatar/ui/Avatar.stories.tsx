import { Meta, StoryObj } from '@storybook/react';
import { Avatar } from '@/shared/ui/Avatar';

const meta: Meta = {
  title: 'UI/Avatar',
  component: Avatar,
};

export default meta;

type Story = StoryObj<typeof Avatar>;

export const Default: Story = {
  render: () => <Avatar />,
};
