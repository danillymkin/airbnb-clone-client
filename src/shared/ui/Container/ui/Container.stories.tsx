import { Meta, StoryObj } from '@storybook/react';
import { Container } from '@/shared/ui/Container';

const meta: Meta = {
  title: 'UI/Container',
  component: Container,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

type Story = StoryObj<typeof Container>;

export const Default: Story = {
  render: () => (
    <Container>
      <div>Children</div>
    </Container>
  ),
};
