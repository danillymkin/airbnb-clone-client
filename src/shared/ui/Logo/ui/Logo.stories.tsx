import { Meta, StoryObj } from '@storybook/react';
import { Logo } from '@/shared/ui/Logo';

const meta: Meta<typeof Logo> = {
  title: 'UI/Logo',
  component: Logo,
};

export default meta;

type Story = StoryObj<typeof Logo>;

export const Default: Story = {
  render: () => <Logo />,
};
