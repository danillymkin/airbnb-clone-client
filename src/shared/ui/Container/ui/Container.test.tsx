import { render, screen } from '@testing-library/react';
import { Container } from '@/shared/ui/Container';

describe('Container', () => {
  it('should render container', () => {
    const children = 'Text';
    render(<Container>{children}</Container>);
    const container = screen.getByText(children);
    expect(container).toBeTruthy();
  });

  it('should match snapshot', () => {
    const children = 'Text';
    render(<Container>{children}</Container>);
    const container = screen.getByText(children);
    expect(container).toMatchSnapshot();
  });

  it('should render children', () => {
    const children = 'Text';
    render(<Container>{children}</Container>);
    const container = screen.getByText(children);
    expect(container).toHaveTextContent(children);
  });
});
