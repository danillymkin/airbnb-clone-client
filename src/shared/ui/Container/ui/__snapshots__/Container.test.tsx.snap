// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match snapshot 1`] = `
<div
  class="mx-auto max-w-[2520px] px-4 sm:px-2 md:px-10 xl:px-20"
>
  Text
</div>
`;
